lib/active_interaction/base.rb

```diff
@@ -163,7 +164,9 @@ def eagerly_evaluate_default(filter)
     def initialize(inputs = {})
       fail ArgumentError, 'inputs must be a hash' unless inputs.is_a?(Hash)
 
-      process_inputs(inputs.symbolize_keys)
+      symbolized_inputs = inputs.symbolize_keys
+      @_interaction_keys = symbolized_inputs.keys.to_set
```

Comment:
Maybe this should only include keys that correspond to filters. As it is, you could sneak some out-of-band information through this.